src/OrleansProviders/Streams/Common/Monitors/ICacheMonitor.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Orleans.Providers.Streams.Common
+{
+    /// <summary>
+    /// Responsible for monitoring cache related metrics
+    /// </summary>
+    public interface ICacheMonitor
+    {
+        #region Event driven metrics
+
+        /// <summary>
+        /// Track cache pressure metrics when cache pressure monitor encounter a status change
+        /// </summary>
+        void TrackCachePressureMonitorStatusChange(string pressureMonitorType, bool underPressure, double? cachePressureContributionCount, double? currentPressure);
+
+        /// <summary>
+        /// Track message added to the cache, and newest and oldest messsges among them 
+        /// </summary>
+        /// <param name="messageAdded"></param>
+        void TrackMessageAdded(long messageAdded);
+
+        /// <summary>
+        /// Track message purged from the cache, and the newest and oldest messages among them
+        /// </summary>
+        /// <param name="messagePurged"></param>
+        void TrackMessagePurged(long messagePurged);
+
+        /// <summary>
+        /// Track new memory allocated by the cache
+        /// </summary>
+        /// <param name="memoryInByte"></param>
+        void TrackMemoryAllocated(int memoryInByte);
```

Comment:
hmm, I think one engineer will understand the cache size doesn't equal to messagecount*messagesize. Since it lies in the concept of cache. When a system allocate a cache, it allocate a size of memory based on the estimated load, instead of actual load. AFAIK, none of the cache providers provide a solution which allocate memory to exact amount of the actual load